crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2372,11 +2372,49 @@ impl<'db> TypeInferenceBuilder<'db> {
             ctx: _,
         } = subscript;
 
-        self.infer_expression(slice);
-        self.infer_expression(value);
+        let value_ty = self.infer_expression(value);
+        let slice_ty = self.infer_expression(slice);
 
-        // TODO actual subscript support
-        Type::Unknown
+        match (value_ty, slice_ty) {
+            // TODO handle variable-length tuples
+            (Type::Tuple(ty), Type::IntLiteral(index)) => {
+                if let Some(ty) = ty.elements(self.db).get(index as usize) {
+                    *ty
+                } else {
+                    Type::Unknown
+                }
+            }
+            (Type::Tuple(ty), Type::BooleanLiteral(bool)) => {
+                if let Some(ty) = ty.elements(self.db).get(usize::from(bool)) {
+                    *ty
+                } else {
+                    Type::Unknown
+                }
+            }
+            (Type::StringLiteral(literal), Type::IntLiteral(index)) => {
+                if let Some(ch) = literal.value(self.db).chars().nth(index as usize) {
+                    Type::StringLiteral(StringLiteralType::new(
+                        self.db,
+                        ch.to_string().into_boxed_str(),
+                    ))
+                } else {
+                    Type::Unknown
+                }
+            }
+            (Type::StringLiteral(literal), Type::BooleanLiteral(index)) => {
+                if let Some(ch) = literal.value(self.db).chars().nth(usize::from(index)) {
+                    Type::StringLiteral(StringLiteralType::new(
+                        self.db,
+                        ch.to_string().into_boxed_str(),
+                    ))
```

Comment:
I assume it is beneficial to compute the `StringLiteral` when we can?

Reply:
yup